File: cpp_pubsub/src/string_subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class StringSubscriber : public rclcpp::Node
{
public:
    StringSubscriber() : Node("string_subscriber")
    {
        subscriber = this->create_subscription<example_interfaces::msg::String>(
            "string_msg", 10,
            std::bind(&StringSubscriber::callbackRobotNews, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "String subscriber started!");
    }

private:
    void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg)
    {
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }

    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<StringSubscriber>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: py_pubsub/py_pubsub/string_subscriber.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
import rclpy.publisher


class StringSubscriber(Node): 
    def __init__(self):
        super().__init__("string_subscriber") 
        self.subscriber = self.create_subscription(String,"my_message",self.stringDataCB,10)
        self.get_logger().info("The Node started!")

    def stringDataCB(self,msg):
        string_data = msg.data
        self.get_logger().info(string_data)


def main(args=None):
    rclpy.init(args=args)
    node = StringSubscriber()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
